imod/msw/sprinkling.py

```diff
@@ -76,31 +75,42 @@ def _render(
         mf6_dis: StructuredDiscretization,
         mf6_well: Mf6Wel,
     ):
+        def ravel_per_subunit(array: xr.DataArray) -> np.ndarray:
+            # per defined well element, all subunits
+            array_out = array.to_numpy()[:, well_row, well_column].ravel()
+            # per defined well element, per defined subunits
+            return array_out[np.isfinite(array_out)]
+
         if not isinstance(mf6_well, Mf6Wel):
             raise TypeError(rf"well not of type 'Mf6Wel', got '{type(mf6_well)}'")
 
         well_cellid = mf6_well["cellid"]
-        if len(well_cellid.coords["dim_cellid"]) != 3:
-            raise TypeError("Coupling to unstructured grids is not supported.")
 
         well_layer = well_cellid.sel(dim_cellid="layer").data
         well_row = well_cellid.sel(dim_cellid="row").data - 1
         well_column = well_cellid.sel(dim_cellid="column").data - 1
 
-        n_subunit = svat["subunit"].size
-
-        well_svat = svat.data[:, well_row, well_column]
-        well_active = well_svat != 0
-
-        # Tile well_layers for each subunit
-        layer = np.tile(well_layer, (n_subunit, 1))
-
-        data_dict = {"svat": well_svat[well_active], "layer": layer[well_active]}
-
-        for var in self._without_subunit:
-            well_arr = self.dataset[var].data[well_row, well_column]
-            well_arr = np.tile(well_arr, (n_subunit, 1))
-            data_dict[var] = well_arr[well_active]
+        max_rate_per_svat = self.dataset["max_abstraction_groundwater"].where(svat > 0)
+        layer_per_svat = xr.full_like(max_rate_per_svat, np.nan)
+        layer_per_svat.values[:, well_row, well_column] = well_layer
+
+        layer_source = ravel_per_subunit(
+            layer_per_svat.where(max_rate_per_svat > 0)
+        ).astype(dtype=np.int32)
+        svat_source_target = ravel_per_subunit(
+            svat.where(max_rate_per_svat > 0)
+        ).astype(dtype=np.int32)
+
+        data_dict = {
+            "svat": svat_source_target,
+            "layer": layer_source,
+            "svat_groundwater": svat_source_target,
+        }
+
+        for var in self._with_subunit:
+            array = self.dataset[var].where(max_rate_per_svat > 0).to_numpy()
```

Comment:
See previous suggestion:

```suggestion
            array = self.dataset[var].where(is_active_per_svat).to_numpy()
```